src/app/modules/auth/register.page.tsx

```diff
@@ -0,0 +1,167 @@
+import { InputCaption, Link, Text } from '@/atomic/atm.typography';
+import { Form, FormControl, FormField, FormItem } from '@/components/ui/form';
+import { authStrings } from './auth.strings';
+import { TextInput } from '@/atomic/atm.text-input';
+import { Button } from '@/atomic/atm.button';
+import authRoutes from './auth.routes';
+import { useForm } from 'react-hook-form';
+import { UserInput } from '@/app/data/graphql/generated';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { registerSchema } from './auth.schema';
+
+interface RegisterFormInput extends UserInput {
```

Comment:
pra mim, esse parece um bom caso de uso pra extender uma outra interface, mas é importante pensar no seguinte: sempre que houver mudança no `UserInput` você quer que essa mudança seja também feita no `RegisterFormInput`? se a resposta for "sim", então extender o `UserInput` está ótimo, se não, melhor criar uma interface dependente.